tests/php/Forms/GridField/GridFieldTest.php

```diff
@@ -716,4 +732,86 @@ public function testAddRequestToURL()
+                'sudoActive' => true,
+                'expected' => false,
+            ],
+            'form-protected-do-protected-sudo-inactive' => [
+                'formProtected' => true,
+                'dataObjectProtected' => true,
+                'sudoActive' => false,
+                'expected' => true,
+            ],
+            'form-protected-do-unprotected-sudo-active' => [
+                'formProtected' => true,
+                'dataObjectProtected' => false,
+                'sudoActive' => true,
+                'expected' => false,
+            ],
+            'form-protected-do-unprotected-sudo-inactive' => [
+                'formProtected' => true,
+                'dataObjectProtected' => false,
+                'sudoActive' => false,
+                'expected' => false,
+            ],
+            'form-unprotected-do-protected-sudo-active' => [
+                'formProtected' => false,
+                'dataObjectProtected' => true,
+                'sudoActive' => true,
+                'expected' => false,
+            ],
+            'form-unprotected-do-protected-sudo-inactive' => [
+                'formProtected' => false,
+                'dataObjectProtected' => true,
+                'sudoActive' => false,
+                'expected' => true,
+            ],
+            'form-unprotected-do-unprotected-sudo-active' => [
+                'formProtected' => false,
+                'dataObjectProtected' => false,
+                'sudoActive' => true,
+                'expected' => false,
+            ],
+            'form-unprotected-do-unprotected-sudo-inactive' => [
+                'formProtected' => false,
+                'dataObjectProtected' => false,
+                'sudoActive' => false,
+                'expected' => false,
+            ],
+        ];
+    }
+
+    /**
+     * Test to see if a GridFieldSudoMode component is added to the GridField in different scenarios
+     *
+     * @dataProvider provideSudoModeShowsCollapsedField
+     */
+    public function testSudoModeShowsCollapsedField(
+        bool $formProtected,
+        bool $dataObjectProtected,
+        bool $sudoActive,
+        bool $expected
+    ): void {
+        if ($sudoActive) {
+            $session = Controller::curr()->getRequest()->getSession();
+            Injector::inst()->get(SudoModeServiceInterface::class)->activate($session);
+        }
+        $form = new Form();
+        if ($formProtected) {
+            $form->requireSudoMode();
+        }
+        $dataList = $dataObjectProtected ? RequiresSudoMode::get() : Team::get();
+        $gridField = new GridField('testfield', dataList: $dataList);
+        $gridField->setForm($form);
+        // call FieldHolder() as it's where the GridFieldSudoMode component will be added
+        $gridField->FieldHolder();
+        $actual = !is_null($gridField->getConfig()->getComponentByType(GridFieldSudoMode::class));
+        $this->assertSame($expected, $actual);
```

Comment:
Instead of `$expected` say `$hasComponent - it's easier to mentally parse what we're actually checking for that way.

```suggestion
        $this->assertSame($hasComponent, !is_null($gridField->getConfig()->getComponentByType(GridFieldSudoMode::class)));
```

Reply:
Alternatively check the rendered result of the `GridField` which is what the test name implies we're checking.

Reply:
I've gone another route and changed the name of the test to `testSudoModeComponentRendered`